cmd/jaeger/internal/integration/badger_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+const (
+	host        = "0.0.0.0"
+	cleanerPort = "9231"
+	cleanerURL  = "/purge"
+	cleanerAddr = "http://" + host + ":" + cleanerPort
+)
+
+type BadgerStorageIntegration struct {
+	E2EStorageIntegration
+	logger *zap.Logger
+}
+
+func (s *BadgerStorageIntegration) initialize(t *testing.T) {
+	s.e2eInitialize(t)
+
+	s.CleanUp = s.cleanUp
+	s.logger = zap.NewNop()
+}
+
+func (s *BadgerStorageIntegration) cleanUp(t *testing.T) {
+	r, err := http.NewRequest(http.MethodPost, cleanerAddr+cleanerURL, nil)
+	require.NoError(t, err)
+
+	client := &http.Client{}
+
+	resp, err := client.Do(r)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+}
+
+func TestBadgerStorage(t *testing.T) {
+	integration.SkipUnlessEnv(t, "badger")
+
+	s := &BadgerStorageIntegration{
+		E2EStorageIntegration: E2EStorageIntegration{
+			ConfigFile: createBadgerCleanerConfig(t),
+			StorageIntegration: integration.StorageIntegration{
+				SkipBinaryAttrs: true,
+				SkipArchiveTest: true,
+
+				// TODO: remove this once badger supports returning spanKind from GetOperations
+				// Cf https://github.com/jaegertracing/jaeger/issues/1922
+				GetOperationsMissingSpanKind: true,
+			},
+		},
+	}
+	defer s.rmBadgerCeanerConfig(t)
+	s.initialize(t)
+	t.Cleanup(func() {
+		s.e2eCleanUp(t)
+	})
+	s.RunAll(t)
+}
+
+func createBadgerCleanerConfig(t *testing.T) string {
+	cmd := exec.Command("../../../../scripts/prepare-badger-integration-tests.py")
+	data, err := cmd.Output()
+	require.NoError(t, err)
+	tempFile := string(data)
+	tempFile = strings.ReplaceAll(tempFile, "\n", "")
+	return tempFile
```

Comment:
Can rm file via t.cleanup here

Reply:
yes, but we will need to do some refactoring in the test. because the function that removes the file is a method of `BadgerStorageIntegration` and the above function is not.

Reply:
it does not need to be a method. `createBadgerCleanerConfig(t)` creates a temp file, it can also register a cleanup hook to remove that file once the test is done. 

An even better approach is to use tempFile functionality of `testing.T`, especially if you refactor to use Go code for enriching the config instead of depending on external Python script.

Reply:
> An even better approach is to use tempFile functionality of testing.T, especially if you refactor to use Go code for enriching the config instead of depending on external Python script.

i tried doing it in Go. it's a lot complicated and in golang.

Reply:
You're sure? You can parse into a dynamic structure
```go
  var config map[interface{}]interface{}
  err := yaml.Unmarshal(data, &config)
  require.NoError(t, err)

  service, ok := config["service"].(map[interface{}]interface{})
  require.True(t, ok)

  // etc, and at some point add your own map

  newData, err := yaml.Marshal(config)
  // write to temp file
```

Reply:
it worked. now we can easily generalize the config for other storages too